### DISCUSSION TITLE
Why can't Add-Type -TypeDefinition use types added by earlier Add-Type?

### DISCUSSION THREAD

**Discussion by Worldwidebrine** [PowerShell/PowerShell#19012] [Q&A]
```
Add-Type -AssemblyName System.Windows.Forms;
Add-Type -TypeDefinition @'
using System.Windows.Forms;
'@;
```
Error says System.Windows.Forms doesn't exist.

What is the best way to import it?

**Comment by jborean93** [comment]
I believe the rules are slightly different between PowerShell 5.1 and 6+ but AFAIK Add-Type only includes the assemblies loaded in pwsh by default. If you wish to add more assemblies to what it can reference you need to use `-ReferencedAssemblies`.

```powershell
Add-Type -ReferencedAssemblies System.Windows.Forms -TypeDefinition @'
using System.Windows.Forms;
'@
```
